SabreTools.Serialization/Wrappers/PortableExecutable.Extraction.cs

```diff
@@ -158,6 +160,97 @@ public bool ExtractCExe(string outputDirectory, bool includeDebug)
+        {
+            try
+            {
+                var overlayAddress = (int)OverlayAddress;
+                
+                // Return if overlay doesn't exist
+                if (overlayAddress == -1) 
+                    return false;
+                
+                // Ensure the stream is starting at the overlay address
+                _dataSource.Seek(overlayAddress, SeekOrigin.Begin);
+
+                var extracting = true;
+                var streamLength = _dataSource.Length;
+                
+                while (extracting)
+                {
+                    lock (_dataSourceLock)
+                    {
+                        // Try to deserialize the source data
+                        var deserializer = new Readers.InstallShieldExecutableFile();
+                        var entry = deserializer.Deserialize(_dataSource);
+                        if (entry == null)
+                            return false;
+                    
+                        // Get the length, and make sure it won't EOF
+                        var length = entry.Length;
+                        var position = _dataSource.Position;
+                    
+                        if (length > streamLength - position)
+                        {
+                            extracting = false;
+                            break;
+                        }
+
+                        // Ensure directory separators are consistent
+                        // Path is used instead of Name because Path contains the filename anyways.
+                        if (entry.Path == null)
+                            return false;
+                    
+                        var filename = entry.Path.TrimEnd('\0');
+                        if (Path.DirectorySeparatorChar == '\\')
+                            filename = filename.Replace('/', '\\');
+                        else if (Path.DirectorySeparatorChar == '/')
+                            filename = filename.Replace('\\', '/');
+
+                        // Ensure the full output directory exists
+                        filename = Path.Combine(outputDirectory, filename);
+                        var directoryName = Path.GetDirectoryName(filename);
+                        if (directoryName != null && !Directory.Exists(directoryName))
+                            Directory.CreateDirectory(directoryName);
+
+                        // Write the output file
+                        var fs = File.Open(filename, FileMode.Create, FileAccess.Write, FileShare.ReadWrite);
+                        Console.WriteLine($"Attempting to extract {entry.Name} from potential InstallShield Executable");
+
+                        // Read from file in chunks in order to save memory, since some extracted files will be large
+                        // Chunk size is purely arbitrary and can be adjusted as needed.
+                        var chunkSize = 65536;
```

Comment:
Primitive types should use that type. e.g. `int chunkSize`.

Reply:
Should be fixed.